src/NHapi.Base/SupportClass.cs

```diff
@@ -1,3268 +1,3486 @@
+        protected IXmlSaxContentHandler callBackHandler;
+        protected IXmlSaxErrorHandler errorHandler;
+        protected XmlSaxLocatorImpl locator;
+        protected IXmlSaxLexicalHandler lexical;
+        protected IXmlSaxEntityResolver entityResolver;
+        protected string parserFileName;
+
+        /// <summary>
+        /// Public constructor for the class.
+        /// </summary>
+        public XmlSAXDocumentManager()
+        {
+            isValidating = false;
+            namespaceAllowed = false;
+            reader = null;
+            callBackHandler = null;
+            errorHandler = null;
+            locator = null;
+            lexical = null;
+            entityResolver = null;
+            parserFileName = string.Empty;
+        }
+
+        /// <summary>
+        /// Returns a new instance of 'XmlSAXDocumentManager'.
+        /// </summary>
+        /// <returns>A new 'XmlSAXDocumentManager' instance.</returns>
+        public static XmlSAXDocumentManager NewInstance()
+        {
+            return new XmlSAXDocumentManager();
+        }
+
+        /// <summary>
+        /// Returns a clone instance of 'XmlSAXDocumentManager'.
+        /// </summary>
+        /// <returns>A clone 'XmlSAXDocumentManager' instance.</returns>
+        public static XmlSAXDocumentManager CloneInstance(XmlSAXDocumentManager instance)
+        {
+            XmlSAXDocumentManager temp = new XmlSAXDocumentManager();
+            temp.NamespaceAllowed = instance.NamespaceAllowed;
+            temp.isValidating = instance.isValidating;
+            IXmlSaxContentHandler contentHandler = instance.getContentHandler();
+            if (contentHandler != null)
+            {
+                temp.setContentHandler(contentHandler);
+            }
+
+            IXmlSaxErrorHandler errorHandler = instance.getErrorHandler();
+            if (errorHandler != null)
+            {
+                temp.setErrorHandler(errorHandler);
+            }
+
+            temp.setFeature(
+                "http://xml.org/sax/features/namespaces",
+                instance.getFeature("http://xml.org/sax/features/namespaces"));
+            temp.setFeature(
+                "http://xml.org/sax/features/namespace-prefixes",
+                instance.getFeature("http://xml.org/sax/features/namespace-prefixes"));
+            temp.setFeature(
+                "http://xml.org/sax/features/validation",
+                instance.getFeature("http://xml.org/sax/features/validation"));
+            temp.setProperty(
+                "http://xml.org/sax/properties/lexical-handler",
+                instance.getProperty("http://xml.org/sax/properties/lexical-handler"));
+            temp.parserFileName = instance.parserFileName;
+            return temp;
+        }
+
+        /// <summary>
+        /// Indicates whether the 'XmlSAXDocumentManager' are validating the XML over a DTD.
+        /// </summary>
+        public bool IsValidating

```

Comment:
kill fields and just use props?

Reply:
will log a separate issue to refactor this `SupportClass`